tests/unit/scanners/test.javascript.js

```diff
@@ -103,6 +103,53 @@ describe('JavaScript Scanner', () => {
     expect(linterMessages.length).toEqual(0);
   });
 
+  it('should support optional chaining', async () => {
+    const code = 'const dogName = adventurer.dog?.name;';
+
+    const jsScanner = new JavaScriptScanner(code, 'code.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
+  });
+
+  it('should support nullish coalescing operator', async () => {
+    const code = 'const baz = 0 ?? 42;';
+
+    const jsScanner = new JavaScriptScanner(code, 'code.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
+  });
+
+  // eslint-disable-next-line jest/no-disabled-tests
+  it.skip('should support public class fields', async () => {
```

Comment:
I think this isnt supported yet but we should start adding some test cases for various syntaxes for the future

Reply:
apparently ([even if Firefox seems to support them from version 69 and Chrome from version 72](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields#Browser_compatibility)) public class instance fields are still stage 3: https://github.com/tc39/proposal-class-fields

and unfortunately I was unable to find an existing issue in the eslint repository that does track it (I did find one closed about the private fields, but none about the public instance fields)

we may just reference https://github.com/tc39/proposal-class-fields here to more quickly double-check when the README in that repo will be changed to stage 4.